src/metabase/models/json_migration.clj

```diff
@@ -0,0 +1,24 @@
+(ns metabase.models.json-migration)
+
+(defn update-version
+  "Set the updated version"
+  [column-value desired-version]
+  (assoc column-value :version desired-version))
+
+(defmacro def-json-migration
```

Comment:
I think this docstring needs more information. When you first see this in the calling namespace, it will feel like a bit of magic. Making this explain the `[current desired]` dispatch values would be nice, explaining the `::identity` or no-op on `[x x]` values, etc.
i also suspect we need to update the clj-kondo config so that it understands this introduces a new defmulti. At the moment I suspect clj-kondo will be confused by a `defmethod` with no parent `defmulti` that it can see.